### PR TITLE
89 edit post

### PR DESCRIPTION
**Pull request by dev-mohmmed-alshorafa**

### CODE REVIEW

File: src/schemes/post.ts [hunk 1]
```diff
@@ -4,8 +4,8 @@ const postSchema = Joi.object({
   image: Joi.string()
     .regex(/(https?:\/\/.*\.(?:png|jpg|jpeg|gif))/i)
     .allow(''),
-  AnimalId: Joi.number().required(),
-  TagId: Joi.number().required(),
+  Animal: { id: Joi.number().required(), name: Joi.string().required() },
```
Comment (AseelL) [inline src/schemes/post.ts:7]: post controllers were designed to take an id only according to the issues instructing how to build them. this is somewhat considered a redesign to that logic that should've been documented by an issue or a comment on this pr at least

File: src/controllers/PostsController.ts [hunk 2]
```diff
@@ -95,27 +95,38 @@ export default class PostsController {
...
-    });
-    const [, [updatePost]] = await Post.update(validatedUpdate, {
-      where: { id: req.params.postId, UserId: req.user?.id },
-      returning: true,
-    });
-    if (!updatePost) {
-      throw new CustomError(400, 'you cannot update this post');
+  public static async update(req: Request, res: Response) {
+    try {
+      const { content, image, Animal, Tag, User } = req.body;
```
Comment (AseelL) [inline src/controllers/PostsController.ts:100]: please revert the backend files to their original state, this is not how we deal with roadblocks in frontend

**Review by AseelL — CHANGES_REQUESTED** [review]

**Review by AseelL — APPROVED** [review]

File: client/src/components/PostCard/EditImagePost.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+          height: '106%',
+          padding: ' 0 10px',
+          display: 'flex',
+          flexDirection: 'column',
+          alignItems: 'flex-end',
+          justifyContent: 'space-between',
+          position: 'absolute',
+          top: ' -8px',
+          right: '-7px',
+        }}
```
Comment (Saeed99Madi) [inline client/src/components/PostCard/EditImagePost.tsx:62]: adding the style by `sx` attribute is a bad practice

File: client/src/components/PostCard/index.tsx [hunk 4]
```diff
@@ -52,27 +53,40 @@ const Post = ({ post }: { post: IPost }) => {
   return (
     <div ref={ref} className="post-card">
       <article className="article abusluot-btns">
-        <section className="post-content-continuer">
+        <section style={{ width: '100%' }} className="post-content-continuer">
```
Comment (Saeed99Madi) [inline client/src/components/PostCard/index.tsx:56]: also here

**Review by Saeed99Madi — APPROVED** [review]

**Review by aaamra — APPROVED** [review]

**Pull request merged**